compiler/modules/nback/number.bal

```diff
@@ -86,6 +86,37 @@ function buildFloatArithmeticBinary(llvm:Builder builder, Scaffold scaffold, bir
     buildStoreFloat(builder, scaffold, result, insn.result);                                  
 }
 
+function buildDecimalArithmeticBinary(llvm:Builder builder, Scaffold scaffold, bir:DecimalArithmeticBinaryInsn insn) {
+    llvm:Value lhs = buildDecimal(builder, scaffold, insn.operands[0]);
+    llvm:Value rhs = buildDecimal(builder, scaffold, insn.operands[1]);
+    llvm:Value resultWithErr = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(decimalArithmaticFunc(insn.op)), [lhs, rhs]);
+    llvm:BasicBlock continueBlock = scaffold.addBasicBlock();
+    llvm:BasicBlock errBlock = scaffold.addBasicBlock();
+    llvm:Value panicCode = builder.extractValue(resultWithErr, 1);
+    builder.condBr(builder.iCmp("ugt", panicCode, llvm:constInt("i64", 0)), errBlock, continueBlock);
+    builder.positionAtEnd(errBlock);
+    builder.store(buildErrorForPanic(builder, scaffold, panicCode, insn.pos), scaffold.panicAddress());
+    builder.br(scaffold.getOnPanic());
+    builder.positionAtEnd(continueBlock);
+    buildStoreDecimal(builder, scaffold, builder.extractValue(resultWithErr, 0), insn.result);                                  
+}
+
+final readonly & map<string> decimalArithmeticFuncName = {
```

Comment:
Put this below (under floatArithmeticOps) and leave out the `decimal_` prefix. Then use this to build at module-init time a map from the operator to the RuntimeFunction.